tests/chainer_tests/functions_tests/math_tests/test_sum.py

```diff
@@ -11,18 +11,29 @@
 from chainer.testing import condition
 
 
+@testing.parameterize(
+    {'dtype': numpy.float16},
+    {'dtype': numpy.float32},
+    {'dtype': numpy.float64},
+)
 class TestSum(unittest.TestCase):
 
     def setUp(self):
-        self.x = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(numpy.float32)
-        self.gy = numpy.random.uniform(-1, 1, ()).astype(numpy.float32)
+        self.x = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
```

Comment:
If you agree that adding more tests regarding ``shape``, this line should be changed like
```python
self.x = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.shape)
```